sdk/src/main/java/com/horizen/cryptolibprovider/FieldElementUtils.java

```diff
@@ -12,11 +12,11 @@ public static int fieldElementLength() {
     }
 
     public static FieldElement messageToFieldElement(byte[] message) {
-        if (message.length > fieldElementLength()) {
+        if (message.length != fieldElementLength()) {
```

Comment:
There are lots of UT failing for this modification. As a matter of fact we do use messages shorter than 32 bytes. I think we should opt for keeping the original code and explicitly 
left-padding byte arrays prior to calling FieldElement.deserialize(), as described in the NCC recommendation paragraph.

Reply:
private def generateHashAndCleanUp(elements: Array[Byte]*): Array[Byte] = {
    val digest = PoseidonHash.getInstanceConstantLength(elements.length)
    elements.foreach { message =>
      if (message.length > fieldElementLength) {
        throw new IllegalArgumentException("Message length is exceed allowed message len. Message len " +
          message.length + " but it shall be less than " + fieldElementLength)
      }
      val fieldElement = FieldElement.deserialize(util.Arrays.copyOf(message, fieldElementLength))
      digest.update(fieldElement)
      fieldElement.freeFieldElement()
    }
    val hash = digest.finalizeHash()
    val result = hash.serializeFieldElement()
    digest.freePoseidonHash()
    hash.freeFieldElement()
    result
  }

Reply:
@alsala so in the place where lenght is shorter than 32, I implemented in place